B018 wrongly detects inline variable or attribute docstrings

Having an inline attribute doc string or a module variable docstring, it is wrongly marked as B018 (sample from sphinx doc):

```
module_level_variable2 = 98765
"""int: Module level variable documented inline.

The docstring may span multiple lines. The type may optionally be specified
on the first line, separated by a colon.
"""

```

I am hitting this issue for format strings used as attribute docstrings.

Note that attribute docstrings are listed as an official convention in [PEP 257](https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring).

> String literals occurring immediately **after a simple assignment at the top level** of a module, **class**, or __init__ method are called "attribute docstrings".

Here is a false positive:

```py
from dataclasses import dataclass

x = 0


@dataclass
class Foo:
    bar: int
    f"""Attribute docstring {x}"""
```

Output:

```
main.py:9:5: B018 Found useless expression. Either assign it to a variable or remove it.
```

@cooperlees Do *format *string checks need to be temporarily rolled back too?

### Environment

- Python 3.9.8

```
pip list
Package        Version
-------------- --------
attrs          21.4.0
click          8.0.1
flake8         4.0.1
flake8-bugbear 21.11.29
mccabe         0.6.1
pep517         0.10.0
pip            21.3.1
pip-tools      6.4.0
pycodestyle    2.8.0
pyflakes       2.4.0
setuptools     57.0.0
toml           0.10.2
wheel          0.36.2
```

--- a/bugbear.py
+++ b/bugbear.py
@@ -657,7 +657,7 @@ class BugBearVisitor(ast.NodeVisitor):
         for index, subnode in enumerate(node.body):
             if not isinstance(subnode, ast.Expr):
                 continue
-            if index == 0 and isinstance(subnode.value, ast.Str):
+            if index == 0 and isinstance(subnode.value, (ast.Str, ast.JoinedStr)):
                 continue  # most likely a docstring
             if isinstance(
                 subnode.value,

--- a/bugbear.py
+++ b/bugbear.py
@@ -657,15 +657,12 @@ class BugBearVisitor(ast.NodeVisitor):
         for index, subnode in enumerate(node.body):
             if not isinstance(subnode, ast.Expr):
                 continue
-            if index == 0 and isinstance(subnode.value, (ast.Str, ast.JoinedStr)):
-                continue  # most likely a docstring
             if isinstance(
                 subnode.value,
                 (
                     ast.Num,
                     ast.Bytes,
                     ast.NameConstant,
-                    ast.JoinedStr,
                     ast.List,
                     ast.Set,
                     ast.Dict,

--- a/bugbear.py
+++ b/bugbear.py
@@ -654,7 +654,7 @@ class BugBearVisitor(ast.NodeVisitor):
         self.errors.append(B903(node.lineno, node.col_offset))
 
     def check_for_b018(self, node):
-        for index, subnode in enumerate(node.body):
+        for subnode in node.body:
             if not isinstance(subnode, ast.Expr):
                 continue
             if isinstance(

--- a/tests/b018_classes.py
+++ b/tests/b018_classes.py
@@ -1,6 +1,6 @@
 """
 Should emit:
-B018 - on lines 16-26, 30, 33
+B018 - on lines 17-26, 30, 33
 """
 
 
@@ -12,12 +12,12 @@ class Foo2:
     """abc"""
 
     a = 2
-    "str"  # Str
+    "str"  # Str (no raise)
+    f"{int}"  # JoinedStr (no raise)
     1j  # Number (complex)
     1  # Number (int)
     1.0  # Number (float)
     b"foo"  # Binary
-    f"{int}"  # JoinedStr
     True  # NameConstant (True)
     False  # NameConstant (False)
     None  # NameConstant (None)
@@ -31,4 +31,3 @@ class Foo3:
     a = 2
     "str"
     1
-    f"should not raise"

--- a/tests/b018_classes.py
+++ b/tests/b018_classes.py
@@ -31,3 +31,4 @@ class Foo3:
     a = 2
     "str"
     1
+    f"should not raise"

--- a/tests/b018_functions.py
+++ b/tests/b018_functions.py
@@ -30,3 +30,4 @@ def foo3():
     a = 2
     "str"
     3
+    f"should not raise"

--- a/tests/b018_functions.py
+++ b/tests/b018_functions.py
@@ -1,6 +1,6 @@
 """
 Should emit:
-B018 - on lines 15-25, 29, 32
+B018 - on lines 16-25, 29, 32
 """
 
 
@@ -11,12 +11,12 @@ def foo1():
 def foo2():
     """my docstring"""
     a = 2
-    "str"  # Str
+    "str"  # Str (no raise)
+    f"{int}"  # JoinedStr (no raise)
     1j  # Number (complex)
     1  # Number (int)
     1.0  # Number (float)
     b"foo"  # Binary
-    f"{int}"  # JoinedStr
     True  # NameConstant (True)
     False  # NameConstant (False)
     None  # NameConstant (None)
@@ -30,4 +30,3 @@ def foo3():
     a = 2
     "str"
     3
-    f"should not raise"

--- a/tests/test_bugbear.py
+++ b/tests/test_bugbear.py
@@ -234,7 +234,7 @@ class BugbearTestCase(unittest.TestCase):
         bbc = BugBearChecker(filename=str(filename))
         errors = list(bbc.run())
 
-        expected = [B018(line, 4) for line in range(15, 26)]
+        expected = [B018(line, 4) for line in range(16, 26)]
         expected.append(B018(29, 4))
         expected.append(B018(32, 4))
         self.assertEqual(errors, self.errors(*expected))
@@ -244,7 +244,7 @@ class BugbearTestCase(unittest.TestCase):
         bbc = BugBearChecker(filename=str(filename))
         errors = list(bbc.run())
 
-        expected = [B018(line, 4) for line in range(16, 27)]
+        expected = [B018(line, 4) for line in range(17, 27)]
         expected.append(B018(30, 4))
         expected.append(B018(33, 4))
         self.assertEqual(errors, self.errors(*expected))

